### PR TITLE
[Merged by Bors] - feat: port CategoryTheory.Arrow 

### PR DESCRIPTION
**Pull request by mattrobball**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/CategoryTheory/Arrow.lean [hunk 1]
```diff
@@ -0,0 +1,336 @@
...
+
+end Functor
+
+/-- The images of `f : Arrow C` by two isomorphic functors `F : C ⥤ D` are
+isomorphic arrows in `D`. -/
+def Arrow.isoOfNatIso {C D : Type _} [Category C] [Category D] {F G : C ⥤ D} (e : F ≅ G)
+    (f : Arrow C) : F.mapArrow.obj f ≅ G.mapArrow.obj f :=
+  Arrow.isoMk (e.app f.left) (e.app f.right) (by simp)
+#align category_theory.arrow.iso_of_nat_iso CategoryTheory.Arrow.isoOfNatIso
+-- #check @CategoryTheory.Arrow.instCoeHomToQuiverToCategoryStructArrow
```
Comment (jcommelin) [inline Mathlib/CategoryTheory/Arrow.lean:334]:
```suggestion
```

File: Mathlib/CategoryTheory/Arrow.lean [hunk 2]
```diff
@@ -0,0 +1,336 @@
...
+/-- We can also build a morphism in the arrow category out of any commutative square in `T`. -/
+@[simps]
+def homMk' {X Y : T} {f : X ⟶ Y} {P Q : T} {g : P ⟶ Q} {u : X ⟶ P} {v : Y ⟶ Q} (w : u ≫ g = f ≫ v) :
+    Arrow.mk f ⟶ Arrow.mk g where
+  left := u
+  right := v
+  w := w
+#align category_theory.arrow.hom_mk' CategoryTheory.Arrow.homMk'
+
+/- Porting note : removred simp from reassoc -/
```
Comment (jcommelin) [inline Mathlib/CategoryTheory/Arrow.lean:112]:
```suggestion
/- Porting note : removed simp from reassoc -/
```
It looks like the `simp` is still there... is the note out of date?

Comment (mattrobball) [reply Mathlib/CategoryTheory/Arrow.lean]: Yes. Corrected it.

**Comment by jcommelin** [comment]
Thanks :tada:

bors merge

**Pull request closed without merging**